packages/@aws-cdk/assertions/lib/matcher.ts

```diff
@@ -83,6 +105,16 @@ export class MatchResult {
       return '' + r.message + loc + ` (using ${r.matcher.name} matcher)`;
     });
   }
+
+  /** @internal */
+  public _registerCapture(capture: Capture, val: any): void {
```

Comment:
This could be generalized somewhat.

```ts
registerSuccessHandler(handler: ISuccessHandler): void
```

Might be useful to other matchers as well.